src/BasePostMessageStream.test.ts

```diff
@@ -14,6 +14,32 @@ describe('BasePostMessageStream', () => {
     stream.destroy();
   });
 
+  it('checks logger receiving messages', async () => {
+    await new Promise<void>((resolve) => {
+      stream._setLogger((data, out) => {
+        expect(out).toStrictEqual(false);
+        expect(data).toStrictEqual({ data: 123 });
+        resolve();
+      });
+
+      (stream as any)._init = true;
+      (stream as any)._onData({ data: 123 });
+    });
+  });
```

Comment:
Instead of using a promise, what do you think about using a Jest function to assert that the logger function got called? Then the assertion step can go last:

``` suggestion
  it('checks logger receiving messages', () => {
    const log = jest.fn();
    stream._setLogger(log);
    (stream as any)._init = true;
    (stream as any)._onData({ data: 123 });
    expect(log).toHaveBeenCalledWith({ data: 123 }, false);
  });